### PR TITLE
Feature/TR-765/load test taker response in audio PCI

### PR DESCRIPTION
**Pull request by jsconan**
Related to: https://oat-sa.atlassian.net/browse/TR-765

Requires: 
 - [x] https://github.com/oat-sa/tao-item-runner-qti-fe/pull/159
 - [x] https://github.com/oat-sa/extension-tao-itemqti/pull/1662
 - [x] https://github.com/oat-sa/extension-tao-itemqti-pci/pull/221

Update the Results page to allow loading the file content related to QTI file variables. This mainly aims the `audioRecording` PCI, but technically it could apply to any interaction working with a file variable.

Basically, here is what changed:
- the Results controller gets an additional endpoint to serve the file content from a QTI variable
- the variable formatter is supplying the variable URI  instead of replacing the file variable by an empty value
- the `viewResults` client controller scrutinizes the response state of the item, searching for partial file response, then it loads the file content thanks to the supplied variable URI and updates the state before opening up the review component

How to test:
- apply the companion PRs
- take a test containing an audio PCI (you may use the one linked in the ticket, caution the first copy I made contained an outdated version of the PCI, be sure to take the updated one, labelled `test_audio_recording_v12.zip`)
- login in the backoffice and go to the results page
- review the item containing the audio PCI, it should allow replaying the record

Edit: The following issue has been fixed.
**Note:** ~~for the time being, a regression exists because of RE-228. This prevents the state to be available for the item review. While the fix is not there, you can workaround this issue by applying this patch:~~
- ~~open `taoOutcomeUi/model/ResultsService.php`~~
- ~~replace the line 608 by `$variablesByItem[$time] = array_merge($variablesByItem[$time]??[], $this->getItemInfos($itemCallId, [[$itemVariable]]));`~~

**Note 2:** for the time being, the audioRecording PCI allows to record in the review mode as well. This will be addressed by MS-988 and is not part of this PR.

### CODE REVIEW

**Review by btamas — COMMENTED** [review]
I have this error on result page. Is it related?
Screen Shot 2021-03-19 at 15 06 19

File: views/js/controller/viewResult.js [hunk 1]
```diff
@@ -101,10 +182,24 @@ define([
                     };
                 }
 
-                previewerFactory(type, uri, state, {
-                    view: 'reviewRenderer',
-                    fullPage: true
-                });
+                Promise.resolve($btn.data('state'))
+                    .then(state => {
+                        console.log(JSON.stringify(state));
```
Comment (btamas) [inline views/js/controller/viewResult.js:187]: Console.log

File: views/js/controller/viewResult.js [hunk 2]
```diff
@@ -101,10 +182,24 @@ define([
...
+                        return state;
+                    })
+                    .then(state => refineItemState(state, deliveryUri))
+                    .then(state => {
+                        console.log(JSON.stringify(state));
```
Comment (btamas) [inline views/js/controller/viewResult.js:192]: one more

**Review by btamas — CHANGES_REQUESTED** [review]

**Comment by jsconan** [comment]
> I have this error on result page. Is it related?
> Screen Shot 2021-03-19 at 15 06 19

Yes, this is related to RE-228, as mentioned in the PR description. I'll check if a fix has been put on develop and I'll merge it

File: views/js/controller/viewResult.js (hunk 1 above)
Comment (jsconan) [reply views/js/controller/viewResult.js]: oh lord! My bad. Fixed

**Comment by jsconan** [comment]
The PHP issue should be fixed now. I merged back develop.

**Comment by btamas** [comment]
> Yes, this is related to RE-228, as mentioned in the PR description. I'll check if a fix has been put on develop and I'll merge it

I've not read that part.

**Review by btamas — APPROVED** [review]
Works fine, but I was not able to check PHP.

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] The feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

**Review by augustas — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by tikhanovichA — APPROVED** [review]
**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] The feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by anshul119 — APPROVED** [review]
Works as expected! 

**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] The feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**